### PR TITLE
Seal private classes

### PR DESCRIPTION
**Pull request by xtqqczze**
Search and replace `private class` -> `private sealed class`.

* Fix [CS0509: Cannot derive from sealed type](https://docs.microsoft.com/dotnet/csharp/misc/cs0509)
* Fix [CS0628: New protected member declared in sealed class](https://docs.microsoft.com/dotnet/csharp/misc/cs0628)

_Contributes to #15110._

### CODE REVIEW

**Comment by xtqqczze** [comment]
CodeFactor new "complex method" issues are false positives, see #15721.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 1]
```diff
@@ -2916,7 +2916,7 @@ private class ConsoleHostStartupException : Exception
             {
             }
 
-            protected
+            internal
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:2919]: Please revert unrelated change.

Comment (xtqqczze) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: Fixed in a573ebf. Protected member cannot exist in sealed class, see [CS0628](https://docs.microsoft.com/dotnet/csharp/misc/cs0628)

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 2]
```diff
@@ -2916,7 +2916,7 @@ private class ConsoleHostStartupException : Exception
             {
             }
 
-            protected
+            private
             ConsoleHostStartupException(
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:2920]: Please fix this in another PR.

Comment (xtqqczze) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:2920]: It is a compiler error, it must be fixed here.

Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:2920]: Because of `sealed`?

Comment (xtqqczze) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:2920]: > A sealed class cannot introduce a protected member because no other class will be able to inherit from the sealed class and use the protected member.

https://docs.microsoft.com/en-gb/dotnet/csharp/misc/cs0628

**Review by iSazonov — APPROVED** [review]
